### ISSUE TITLE
Bunch of CIS Failures for /etc/kubernetes on RancherOS

### ISSUE THREAD

**Issue by ekristen** [rancher/rancher#26396]
**What kind of request is this (question/bug/enhancement/feature request):**

bug/question

**Steps to reproduce (least amount of steps as possible):**

1. Have your control node be on RancherOS
2. Run CIS Permissive Scan

**Result:**

3. Failures for a bunch of files that do not exist in `/etc/kubernetes`

**Other details that may be helpful:**

Unsure if the remediation information is wrong, or the tests are being performed incorrectly, but since almost all of this information resides under the /opt/rke, it's definitely difficult to tell.

All worker nodes in the custom cluster provisioned by rke don't fail, only the original control node.

![Screen Shot 2020-03-31 at 8 12 09 AM](https://user-images.githubusercontent.com/48329/78036383-7742b280-7327-11ea-85e4-a727995b6f17.png)

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.4.0-rc16
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom RKE (not imported)
- Machine type (cloud/VM/metal) and specifications (CPU/memory): Metal
- Kubernetes version (use `kubectl version`):

```
1.15.5
```

gz#9917

**Comment by sowmyav27** [comment]
**Verified on `rancher:2.4.5`** 

- Deploy a 3 node - 1 etcd, 1 control and 1 worker nodes - ROS 1.5.6 version
- Run CIS 1.4 Permissive scan on the cluster after the cluster is `Active`
- 13 tests are reported failures
<img width="1510" alt="Screen Shot 2020-06-25 at 3 34 24 PM" src="https://user-images.githubusercontent.com/26032343/85802230-62dd3e80-b6f9-11ea-9ff3-a8dbf44d8c7e.png">

**Comment by ekristen** [comment]
Issue still present in 2.4.6

**Comment by Ellenqs** [comment]
Environment information:

Rancher version (rancher/rancher/rancher/server image tag or shown bottom left in the UI): 2.5.4-rc8
Installation option (single install/HA): single install

Cluster information:

Cluster type (Hosted/Infrastructure Provider/Custom/Imported): AWS

Steps:
- Deploy a 4 node - 1 etcd, 1 control and 2 worker nodes - ROS 1.5.7 version
- Run CIS 1.5 Permissive scan on the cluster after the cluster is Active
- 9 tests are reported failures

![Screen Shot 2021-01-04 at 11 58 29 AM](https://user-images.githubusercontent.com/75277058/103574453-cbfdf580-4e84-11eb-93e0-ac99d707044f.png)

**Comment by prachidamle** [comment]
This can be tested by creating a CIS v2 Custom Benchmark.
Changes needed to the CIS 1.5 Permissive benchmark config files are:
- In config.yaml and all the tests in the various yaml files replace `/etc/kubernetes/ssl/` with `/opt/rke/etc/kubernetes/ssl/`
- In etcd.yaml file for test# 1.1.11, replace `/node/var/lib/etcd` with `/node/opt/rke/var/lib/etcd`
- Then upload this custom benchmarks files as a configmap according to the docs to create custom benchmark on CIS v2

**Comment by jiaqiluo** [comment]
We hit a similar issue with monitoring v2, and what we did is to allow users to change the directory of certs in the UI like the following screen indicates. 

<img width="713" alt="Screen Shot 2021-01-07 at 3 07 58 PM" src="https://user-images.githubusercontent.com/6218999/103950762-b1cd4d00-50fa-11eb-9e54-e9b8c5938a46.png">

We can consider exposing the value of the ssh directory in both UI and the chart. Is it doable for CIS v2? @prachidamle

**Comment by prachidamle** [comment]
@jiaqiluo No we do not control the CIS benchmark configs - they are mostly set upstream. That is why we have custom benchmarks now to support custom configuration.

**Comment by jiaqiluo** [comment]
@prachidamle  Oh ok, I got it. Then we don't need to change our UI or chart. 😄

**Comment by slickwarren** [comment]
I was able to verify this fix on rancher v2.5.5.
**Steps taken to fix:**
**Configure Cluster and create rancherOS compatible configmap**
* install rancher v2.5.5
* provision rancherOS cluster 
* download kubeconfig for your rancherOS cluster and `export KUBECONFIG=<path-to-config>`
<img width="397" alt="Screen Shot 2021-01-14 at 11 10 45 AM" src="https://user-images.githubusercontent.com/16691014/104637116-33702e00-5659-11eb-80d2-4515a2f0d33c.png">

* go to cluster Explorer -> apps for your rancherOS cluster and install `CIS Benchmark`
* __the following steps are done outside of the rancher server, likely on your local machine__
* download `rke-cis-1.6-permissive` or 1.5 from the [security-scan github page](https://github.com/rancher/security-scan/tree/master/package/cfg) as well as the `config.yaml` file in the link
* using the config.yaml from `rke-cis-1.6-permissive` or equivalent, replace each value from this config into the config.yaml from root of [security-scan github page](https://github.com/rancher/security-scan/tree/master/package/cfg) downloaded in previous step
* remove the config.yaml from the folder. replace it with the root config.yaml with the new overrides
* do a search-and-replace of `/etc/kubernetes/ssl/` with `/opt/rke/etc/kubernetes/ssl/` for _each file_ in the downloaded folder
* do a search-and-replace of `/node/var/lib/etcd` with `/node/opt/rke/var/lib/etcd` for the `etcd.yaml` file in the downloaded folder. 
* remove all but 1 entry of `version mapping` and `target mapping`.  rename the value to any name you desire. I used `rke-1.15`:`ros153`:

```
version_mapping:
  "rke-1.15": "ros153"
target_mapping:
  "ros153":
    - "master"
    - "node"
    - "controlplane"
    - "etcd"
    - "policies"
```
* rename folder to the name you chose from above `ros153`
* create a new configmap for RancherOS compatible CIS test by running `kubectl create configmap -n default <ros153> --from-file=<security-scan/package/cfg/ros153>` replacing your value name and file path appropriately. You can also configure the namespace <default> to whichever namespace you desire running on your cluster

**Configure CIS Benchmark for new configmap** 
* in your rancher server, go to cluster explorer -> CIS Benchmark -> Benchmark Version -> create 
<img width="1789" alt="Screen Shot 2021-01-14 at 11 31 38 AM" src="https://user-images.githubusercontent.com/16691014/104639667-87304680-565c-11eb-8e73-14cd7b52c889.png">

* give the custom benchmark a name, select `rke` as the cluster provider and find your namespace/configmap we configured previously in the `custom benchmark configmap` dropdown option
* now, you can create a profile for this new benchmark, and scan as you would for an auto-configured CIS test
